screenpipe-app-tauri/components/search-history-sidebar.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button } from "@/components/ui/button";
+import { ScrollArea } from "@/components/ui/scroll-area";
+import { cn } from "@/lib/utils";
+import { Plus, Trash2, History } from "lucide-react";
+import { formatDistanceToNow } from "date-fns";
+import { SearchHistory } from "@/lib/types/history";
+
+
+interface SearchHistorySidebarProps {
+  searches: SearchHistory[];
+  currentSearchId: string | null;
+  onSelectSearch: (id: string) => void;
+  onDeleteSearch: (id: string) => void;
+  isCollapsed: boolean;
+  onToggleCollapse: () => void;
+  onNewSearch?: () => void;
+}
```

Comment:
wait that does not seem to use the `<Sidebar>` from shadcn? also seems duplicate sidebar

it's better to use their  component which uses best practices in terms of style and maintainability

Reply:
I tried that first. But since it request wrapping a provider. It was throwing some hydration error. I just had to go with this. 

NB: you'll notice a sidebar component in `UI` folder. Which shows my initial efforts on Shadcn.